applications/ParticleMechanicsApplication/custom_utilities/pqmpm_partition_utilities.cpp

```diff
@@ -0,0 +1,795 @@
+                    rPointVector[2][1] += SideHalfLength;
+                    rPointVector[3][1] += SideHalfLength;
+                }
+                else
+                {
+                    if (rPointVector.size() != 8) rPointVector.resize(8);
+                    for (size_t i = 0; i < 8; ++i) {
+                        rPointVector[i].clear();
+                        rPointVector[i] += rCenter;
+                    }
+                    rPointVector[0][0] -= SideHalfLength;
+                    rPointVector[1][0] += SideHalfLength;
+                    rPointVector[2][0] += SideHalfLength;
+                    rPointVector[3][0] -= SideHalfLength;
+                    rPointVector[4][0] -= SideHalfLength;
+                    rPointVector[5][0] += SideHalfLength;
+                    rPointVector[6][0] += SideHalfLength;
+                    rPointVector[7][0] -= SideHalfLength;
+
+                    rPointVector[0][1] -= SideHalfLength;
+                    rPointVector[1][1] -= SideHalfLength;
+                    rPointVector[2][1] += SideHalfLength;
+                    rPointVector[3][1] += SideHalfLength;
+                    rPointVector[4][1] -= SideHalfLength;
+                    rPointVector[5][1] -= SideHalfLength;
+                    rPointVector[6][1] += SideHalfLength;
+                    rPointVector[7][1] += SideHalfLength;
+
+                    rPointVector[0][2] -= SideHalfLength;
+                    rPointVector[1][2] -= SideHalfLength;
+                    rPointVector[2][2] -= SideHalfLength;
+                    rPointVector[3][2] -= SideHalfLength;
+                    rPointVector[4][2] += SideHalfLength;
+                    rPointVector[5][2] += SideHalfLength;
+                    rPointVector[6][2] += SideHalfLength;
+                    rPointVector[7][2] += SideHalfLength;
+                }
+
+            KRATOS_CATCH("")
+        }
+
+
+        bool CheckAllPointsAreInGeom(
+            const std::vector<array_1d<double, 3>>& rPoints,
+            const GeometryType& rReferenceGeom,
+            const double Tolerance)
+        {
+            KRATOS_TRY
+
+                array_1d<double, 3> dummy_local_coords;
+            for (size_t i = 0; i < rPoints.size(); ++i) {
+                if (!MPMSearchElementUtility::CheckIsInside(rReferenceGeom, dummy_local_coords, rPoints[i], Tolerance, false)) {
+                    return false;
+                }
+            }
+            return true;
+
+            KRATOS_CATCH("")
+        }
+
+        void Check3DBackGroundMeshIsCubicAxisAligned(const std::vector<GeometryType*>& rIntersectedGeometries)
+        {
+            KRATOS_TRY
+
+                NodeType point_low, point_high;
+            for (size_t i = 0; i < rIntersectedGeometries.size(); ++i) {
+                if (rIntersectedGeometries[i]->GetGeometryType() != GeometryData::Kratos_Hexahedra3D8) {
+                    #pragma omp single
```

Comment:
It's single so the subsequent error message will be printed during a parallel run. Do you guys have a typically solution/approach for this?

Reply:
there is none with OpenMP. Only when we eventually switch to C++ based parallelism we will get proper error messages

I still recommend removing it as then it would be consistent with the Core, but I will not enforce :)